src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -76,15 +24,15 @@ public static IConfigurationBuilder AddUserSecrets(this IConfigurationBuilder co
                 throw new ArgumentNullException(nameof(configuration));
             }
 
-            var projectPath = configuration.GetProjectPath();
-            if (string.IsNullOrEmpty(projectPath))
+            var fileProvider = configuration.GetFileProvider();
+            if (fileProvider == null)
             {
                 throw new InvalidOperationException(Resources.FormatError_MissingBasePath(
-                        projectPath,
-                        typeof(IConfigurationBuilder).Name,
-                        ProjectPathKey));
+                        nameof(IConfigurationBuilder),
+                        nameof(IFileProvider)));
             }
-            return AddSecretsFile(configuration, PathHelper.GetSecretsPath(projectPath));
+
```

Comment:
This is a pretty ugly looking line, and assuming a FileProvider has been set on the config builder, PathHelper is using File.Exists/Read instead of going through the FileProvider which seems wrong to mix isn't it?